### PR TITLE
Sonalfront

### PR DESCRIPTION
**Pull request by Sonal2808**
Hello. check frontpage.

### CODE REVIEW

File: roomranger/frontend/public/images/Sonal.jpeg
Comment (LunaLiu4) [inline roomranger/frontend/public/images/Sonal.jpeg:1]: I don't think you want to upload this shinning lady photo to of app. hahaha

File: roomranger/frontend/public/images/cat.jpeg
Comment (LunaLiu829) [inline roomranger/frontend/public/images/cat.jpeg:1]: Delete this one please

File: roomranger/frontend/src/App.js [hunk 1]
```diff
@@ -1,49 +1,76 @@
...
-    {/* <Footer/> */}
-    </div>
+        <Container fluid>
+          <Row>
+            <Col md={6}>
+              <Routes>
+                <Route exact path="/registration" element={<Registration />} />
+                <Route exact path="/loginPage" element={<LoginPage />} />
+                <Route exact path="/about" element={<About />} /> 
+                <Route exact path="/" element={<Home />} />
```
Comment (LunaLiu829) [inline roomranger/frontend/src/App.js]: This App.js changed a lot

File: roomranger/frontend/public/images/cat.jpeg
Comment (Sonal2808) [reply roomranger/frontend/public/images/cat.jpeg:1]: Done

File: roomranger/frontend/src/App.js (hunk 1 above)
Comment (Sonal2808) [reply roomranger/frontend/src/App.js]: Made changes as per main

File: roomranger/.gitignore [hunk 2]
```diff
@@ -35,5 +34,5 @@ out/
 /.nb-gradle/
 
 ### VS Code ###
-.vscode/
+../vscode/
```
Comment (LunaLiu829) [inline roomranger/.gitignore:37]: Any reason for this one change?

File: roomranger/build.gradle [hunk 3]
```diff
@@ -2,7 +2,7 @@ plugins {
 	id 'java'
   	id 'org.springframework.boot' version '3.2.0'
 	id 'io.spring.dependency-management' version '1.1.4'
-//	id "com.github.node-gradle.node" version "7.0.1"
+	id "com.github.node-gradle.node" version "7.0.1"
```
Comment (LunaLiu829) [inline roomranger/build.gradle:5]: I feel like we don't need the node gradle for our project , so in last pull I deleted this one.

File: roomranger/frontend/src/App.js [hunk 4]
```diff
@@ -1,81 +1,95 @@
-import './App.css';
-import "../node_modules/bootstrap/dist/css/bootstrap.min.css"
+import "./App.css";
```
Comment (LunaLiu829) [inline roomranger/frontend/src/App.js:1]: Hmm, to maintain consistency, let's all use single quotation marks.

File: roomranger/frontend/src/App.js [hunk 5]
```diff
@@ -1,81 +1,95 @@
...
+                  {/* <Route element={<RouteHome />}>
+                      <Route path='/' element={<Home />} />
+                      <Route path='login' element={<LoginPage />} />
+                      <Route path='register' element={<Register />} />
+                   </Route> */}
+                  {/* </Route> */}
+                  <Route path='/' element={<RouteHome />}>
+                    <Route index element={<Home />} />
+                    <Route path='Register' element={<Registration />} />
+                    <Route path='About' element={<About />} />
```
Comment (LunaLiu829) [inline roomranger/frontend/src/App.js:58]: About will be imported to Footer component, not here

File: roomranger/frontend/src/components/Header.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
+// import React from 'react'
```
Comment (LunaLiu829) [inline roomranger/frontend/src/components/Header.js:1]: We are not using this one, I think you can delete it. Keep it the same as main

File: roomranger/frontend/src/components/NavBar.js [hunk 7]
```diff
@@ -1,30 +1,32 @@
 import React from 'react'
-import { Link, Outlet } from 'react-router-dom';
+import { Link } from 'react-router-dom';
```
Comment (LunaLiu829) [inline roomranger/frontend/src/components/NavBar.js:2]: This file is not the same as main. You are not supposed to change this file.

File: roomranger/frontend/src/components/leaveRequest/home.js [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
```
Comment (LunaLiu829) [inline roomranger/frontend/src/components/leaveRequest/home.js:1]: Delete this file.

File: roomranger/frontend/src/components/loginPage/LoginPage.js [hunk 9]
```diff
@@ -27,7 +27,7 @@ const navigate=useNavigate();
       if(username==='samata' && password==='karka'){
       setShowSuccessMessage(true);
       setShowErrorMessage(false)
-      navigate('/landing')
+      navigate('/login/ManagerHomePage')
```
Comment (LunaLiu829) [inline roomranger/frontend/src/components/loginPage/LoginPage.js:30]: You are not supposed to change this file.. please keep the same as main

File: roomranger/frontend/src/components/manager/AssignRoomForm.js [hunk 10]
```diff
@@ -300,7 +300,7 @@ export default function AssignRoom() {
               </select>
             </div>
             <button type='submit' className='btn btn-outline-primary'>Submit</button>
-            <Link className='btn btn-outline-danger mx-2' to="/landing">Cancel</Link>
+            <Link className='btn btn-outline-danger mx-2' to="/manager">Cancel</Link>
```
Comment (LunaLiu829) [inline roomranger/frontend/src/components/manager/AssignRoomForm.js:303]: Revert your change to keep the same with main

File: roomranger/frontend/src/components/manager/EditAssignedRoom.js [hunk 11]
```diff
@@ -2,8 +2,6 @@ import axios from 'axios';
 import React, { useEffect, useState } from 'react'
```
Comment (LunaLiu829) [inline roomranger/frontend/src/components/manager/EditAssignedRoom.js:2]: You are not supposed to change this file.

File: roomranger/frontend/src/components/manager/HomeManager.js [hunk 12]
```diff
@@ -40,7 +40,7 @@ export default function HomeManager() {
     return (
         <div className='container'>
             <div className='py=4'>
-                <Link className='btn btn-primary' to='/landing/assignroomform'>Assign Room</Link>
+                <Link className='btn btn-primary' to='/assignedrooms/assignroomform'>Assign Room</Link>
```
Comment (LunaLiu829) [inline roomranger/frontend/src/components/manager/HomeManager.js:43]: Same, don't change this file.

File: roomranger/frontend/src/components/roomAttendants/AddAttendant.js [hunk 13]
```diff
@@ -29,7 +29,7 @@ export default function AddAttendant() {
     e.preventDefault();
     try{
     await axios.post("http://localhost:8080/roomAttendant/add", attendant);
-    navigate("/landing/attendants")}
+    navigate("/")}
```
Comment (LunaLiu829) [inline roomranger/frontend/src/components/roomAttendants/AddAttendant.js:32]: Same, don't change this file.

File: roomranger/frontend/src/components/roomAttendants/AttendantListComponent.js [hunk 14]
```diff
@@ -61,15 +61,15 @@ export default function AttendantListComponent() {
                 <td>{roomAttendant.firstName}{' '}{roomAttendant.lastName}</td>
                 <td>{roomAttendant.email}</td>
                 <td >{roomAttendant.notes}</td>
-                <td align='center'><Link className="btn btn-outline-success mx-2" to={`/landing/attendants/profile/${roomAttendant.id}`}>View</Link>
-        <Link className="btn btn-outline-primary mx-2 "to={`/landing/attendants/update/${roomAttendant.id}`} >EDIT</Link>        
+                <td align='center'><Link className="btn btn-outline-success mx-2" to={`/profile/${roomAttendant.id}`}>View</Link>
```
Comment (LunaLiu829) [inline roomranger/frontend/src/components/roomAttendants/AttendantListComponent.js:64]: Same, don't change this file.

File: roomranger/frontend/src/components/roomAttendants/Profile.js [hunk 15]
```diff
@@ -108,7 +108,7 @@ export default function Profile() {
                 </ul>
             </div>
         </div>
-        <Link className='btn btn-primary my-2' to={'/landing/attendants'}>Back to Home</Link>
+        <Link className='btn btn-primary my-2' to={'/'}>Back to Home</Link>
```
Comment (LunaLiu829) [inline roomranger/frontend/src/components/roomAttendants/Profile.js:111]: Same, don't change this file.

File: roomranger/frontend/src/components/roomAttendants/Update.js [hunk 16]
```diff
@@ -34,7 +34,7 @@ useEffect(()=> {
   const onSubmit = async (e) => {
     e.preventDefault();
     await axios.put(`http://localhost:8080/roomAttendant/update/${id}`, attendants);
-    navigate("/landing/attendants")};
+    navigate("/")};
```
Comment (LunaLiu829) [inline roomranger/frontend/src/components/roomAttendants/Update.js:37]: Same, don't change this file.

File: roomranger/frontend/src/components/rooms/AddRoom.js [hunk 17]
```diff
@@ -8,7 +8,7 @@ export default function AddRoom() {
     const [room, setRoom] = useState({
         roomNumber: "",
         roomType: "",
-        available: true
+        available: false
```
Comment (LunaLiu829) [inline roomranger/frontend/src/components/rooms/AddRoom.js:11]: Same, don't change this file.

File: roomranger/frontend/src/components/rooms/EditRoom.js [hunk 18]
```diff
@@ -9,7 +9,7 @@ export default function EditRoom() {
     const [room, setRoom] = useState({
         roomNumber: "",
         roomType: "",
-        available: true
+        available: false
```
Comment (LunaLiu829) [inline roomranger/frontend/src/components/rooms/EditRoom.js:12]: Same, don't change this file.

File: roomranger/frontend/src/components/rooms/HomeRoom.js [hunk 19]
```diff
@@ -1,11 +1,11 @@
 import React, { useEffect, useState } from 'react';
 import axios from 'axios';
-import { Link, useParams} from 'react-router-dom';
+import { Link, useParams } from 'react-router-dom';
```
Comment (LunaLiu829) [inline roomranger/frontend/src/components/rooms/HomeRoom.js:3]: Same, don't change this file.

File: roomranger/frontend/src/components/rooms/ViewRoom.js [hunk 20]
```diff
@@ -1,6 +1,6 @@
 import axios from 'axios';
 import React, { useEffect, useState } from 'react'
-import { Link, useParams } from 'react-router-dom';
+import { Link, useParams } from 'react-router-dom'
```
Comment (LunaLiu829) [inline roomranger/frontend/src/components/rooms/ViewRoom.js:3]: Same, don't change this file.

File: roomranger/src/main/java/org/launchcode/roomranger/controllers/LeaveRequestController.java [hunk 21]
```diff
@@ -1,94 +1,34 @@
...
-import org.launchcode.roomranger.data.LeaveRequestRepository;
-import org.launchcode.roomranger.data.RoomAttendantRepository;
-import org.launchcode.roomranger.exception.NotFoundException;
-import org.launchcode.roomranger.models.LeaveRequest;
-import org.launchcode.roomranger.models.Room;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.data.domain.Sort;
-import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.*;
+import org.launchcode.roomranger.models.Leave;
```
Comment (LunaLiu829) [inline roomranger/src/main/java/org/launchcode/roomranger/controllers/LeaveRequestController.java:3]: Same, don't change this file.

File: roomranger/src/main/java/org/launchcode/roomranger/controllers/RoomsController.java [hunk 22]
```diff
@@ -2,10 +2,11 @@
 
 import jakarta.validation.Valid;
 import org.launchcode.roomranger.data.CommentRepository;
-import org.launchcode.roomranger.data.RoomAttendantRepository;
 import org.launchcode.roomranger.exception.NotFoundException;
 import org.launchcode.roomranger.data.RoomRepository;
-import org.launchcode.roomranger.models.*;
+import org.launchcode.roomranger.models.Comment;
```
Comment (LunaLiu829) [inline roomranger/src/main/java/org/launchcode/roomranger/controllers/RoomsController.java:7]: Same, don't change this file.

File: roomranger/src/main/java/org/launchcode/roomranger/data/CommentRepository.java [hunk 23]
```diff
@@ -1,13 +1,9 @@
...
-import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.repository.CrudRepository;
 import org.springframework.stereotype.Repository;
 
-import java.util.List;
-
 @Repository
-public interface CommentRepository extends JpaRepository<Comment, Integer> {
-    List<Comment> findAllByAssignedRoomId(int assignedRoomId);
+public interface CommentRepository extends CrudRepository<Comment, Integer> {
```
Comment (LunaLiu829) [inline roomranger/src/main/java/org/launchcode/roomranger/data/CommentRepository.java:8]: Same, don't change this file.

File: roomranger/src/main/java/org/launchcode/roomranger/models/AssignedRoom.java [hunk 24]
```diff
@@ -35,8 +33,6 @@ public class AssignedRoom extends AbstractEntity {
 //    @NotNull(message = "Task is Required")
```
Comment (LunaLiu829) [inline roomranger/src/main/java/org/launchcode/roomranger/models/AssignedRoom.java:33]: Same, don't change this file.

File: roomranger/src/main/java/org/launchcode/roomranger/models/Comment.java [hunk 25]
```diff
@@ -1,54 +1,42 @@
 package org.launchcode.roomranger.models;
 
-import com.fasterxml.jackson.annotation.JsonIgnore;
-import jakarta.persistence.*;
+import jakarta.persistence.Column;
```
Comment (LunaLiu829) [inline roomranger/src/main/java/org/launchcode/roomranger/models/Comment.java:3]: Same, don't change this file.

File: roomranger/src/main/java/org/launchcode/roomranger/models/Leave.java [hunk 26]
```diff
@@ -0,0 +1,20 @@
+package org.launchcode.roomranger.models;
```
Comment (LunaLiu829) [inline roomranger/src/main/java/org/launchcode/roomranger/models/Leave.java:1]: Delete this one

File: roomranger/src/main/java/org/launchcode/roomranger/models/Manager.java [hunk 27]
```diff
@@ -13,8 +13,8 @@
 public class Manager extends AbstractEntity{
     private String firstName;
     private String lastName;
-//    @OneToMany(mappedBy = "manager")
-//    private List<RoomAttendant> roomAttendants = new ArrayList<>();
+    @OneToMany(mappedBy = "manager")
```
Comment (LunaLiu829) [inline roomranger/src/main/java/org/launchcode/roomranger/models/Manager.java:16]: Same, don't change this file.

File: roomranger/src/main/java/org/launchcode/roomranger/models/Room.java [hunk 28]
```diff
@@ -17,14 +17,52 @@ public class Room extends AbstractEntity{
     @OneToOne(mappedBy = "room", cascade = CascadeType.ALL)
     @OnDelete(action = OnDeleteAction.CASCADE)
     private AssignedRoom assignedRoom;
+
```
Comment (LunaLiu829) [inline roomranger/src/main/java/org/launchcode/roomranger/models/Room.java:20]: Same, don't change this file.

File: roomranger/src/main/java/org/launchcode/roomranger/models/RoomAttendant.java [hunk 29]
```diff
@@ -46,13 +45,21 @@ public class RoomAttendant  extends AbstractEntity{
 //    public String getWorkingDays() {
 //        return workingDays;
 //    }
-
-    private String workingDays ;
+//
```
Comment (LunaLiu829) [inline roomranger/src/main/java/org/launchcode/roomranger/models/RoomAttendant.java:48]: Same, don't change this file.

File: roomranger/src/main/resources/templates/attendant/leave_request.html [hunk 30]
```diff
@@ -92,100 +91,4 @@
...
-            class="btn btn-primary w-100 shadow-sm">Submit
-            </button>
-        </form>
-    </section>
-</div>
-<script type="text/javascript" th:src="@{/script.js}"></script>
-</body>
-</html>
->>>>>>> react-front-page-luna
+<!--</html>-->
```
Comment (LunaLiu829) [inline roomranger/src/main/resources/templates/attendant/leave_request.html:94]: Same, don't change this file.

File: roomranger/.gitignore (hunk 2 above)
Comment (Sonal2808) [reply roomranger/.gitignore:37]: its just a mistake.Please ignore

File: .gitignore [hunk 31]
```diff
@@ -1,39 +1,2 @@
-HELP.md
-.gradle
-build/
```
Comment (LunaLiu829) [inline .gitignore:3]: Same, don't change this file.

**Pull request closed without merging**